src/librados/AioCompletionImpl.h

```diff
@@ -85,10 +85,7 @@ struct librados::AioCompletionImpl {
     return wait_for_complete();
   }
   int is_complete() {
-    lock.Lock();
-    int r = complete;
-    lock.Unlock();
-    return r;
+    return complete;
```

Comment:
Hi, kefu
	Thanks for you time to check on this. I read your comments but I have different opinions. 
I know we use mutex lock to protect the execute order of finish() function. Especially the order of these code “c->rval = r; c->complete = true; “
Mutex lock will cause thread reschedule, it has memory barrier internal. But this lock is too weight. 
Intel X86 CPU is strong order. These two lines “c->rval = r; c->complete = true;” are two store instructions( refer to this https://en.m.wikipedia.org/wiki/Memory_ordering  store/store is ordered). “c->complete = true” must be executed after “c->rval = r; “ So I think on Intel x86 CPU, there is no need to add the Mutex lock, this will cause performance issue.
But ARM CPU is weak order, “c->complete = true” may be executed before” c->rval = r; “ Without the lock, if another CPU execute these code, such as” if (is_complete() == true) 	{ int retVal = get_return_value()} “ this retVal may be the old value, it is incorrect, because “c->complete = true;” is executed before “c->rval = r;” 
We can use a memory barrier after “c->rval = r;” to keep this code is executed before “c->complete = true”, it is much better than mutex lock.
Thanks.